### PR TITLE
SQLite3: Add collation support for string and text columns

### PR DESCRIPTION
**Pull request by akshay-vishnoi**
Related with #19549
@rafaelfranca  - Please have a look at it.

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb [hunk 1]
```diff
@@ -86,6 +86,11 @@ def add_column_options!(sql, options)
             if options[:primary_key] == true
               sql << " PRIMARY KEY"
             end
+
+            if options[:collation].present?
```
Comment (rafaelfranca) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb]: Do SQL Server and Oracle have support to collation? If not we should not move to abstract adapter.

Comment (yahonda) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb]: As far as I know Oracle database does not have collation support.

Comment (rafaelfranca) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb]: @akshay-vishnoi So we should move it back to each adapter individually.

**Comment by rafaelfranca** [comment]
@akshay-vishnoi thank you for the pull request and really sorry for not merging yours before.

**Comment by akshay-vishnoi** [comment]
@rafaelfranca - No problem. I have updated the PR please check it now.

**Comment by akshay-vishnoi** [comment]
**Rebase** done!!! :smile:

**Comment by akshay-vishnoi** [comment]
@rafaelfranca -  Rebase done again. I think changelog will create conflicts again. Please let me know when you review the PR i will rebase it then if there will be conflicts. :smile:

File: activerecord/test/cases/adapters/sqlite3/sqlite3_adapter_test.rb [hunk 2]
```diff
@@ -421,14 +421,14 @@ def test_respond_to_disable_extension
...
                                    configurations['arunit']['database'])
-        statement = SQLite3::Statement.new(db,
+        statement = ::SQLite3::Statement.new(db,
                                            'CREATE TABLE statement_test (number integer not null)')
-        statement.stubs(:step).raises(SQLite3::BusyException, 'busy')
+        statement.stubs(:step).raises(::SQLite3::BusyException, 'busy')
         statement.stubs(:columns).once.returns([])
         statement.expects(:close).once
-        SQLite3::Statement.stubs(:new).returns(statement)
+        ::SQLite3::Statement.stubs(:new).returns(statement)
```
Comment (vipulnsward) [inline activerecord/test/cases/adapters/sqlite3/sqlite3_adapter_test.rb]: Were these needed?

**Pull request merged**